course/src/Vector3.cc

```diff
@@ -0,0 +1,168 @@
+#include "Vector3.h"
+
+namespace ekumen
+{
+
+Vector3::Vector3(const double x, const double y, const double z)
+{
+    values[0] = x;
+    values[1] = y;
+    values[2] = z;
+}
+Vector3::Vector3()
+{
+    values[0] = 0.;
+    values[1] = 0.;
+    values[2] = 0.;
+}
+Vector3::Vector3(const std::initializer_list<double> &list)
+{
+    int i = 0;
+    for (const auto &v : list)
+    {
+        values[i] = v;
+        i++;
+    }
+}
+
+Vector3::Vector3(const Vector3 &p2)
+{
+    x() = p2.x();
+    y() = p2.y();
+    z() = p2.z();
+}
+
+Vector3::~Vector3() {}
+
+double Vector3::norm() const
+{
+    return sqrt(dot(*this));
+}
+
+double Vector3::dot(const Vector3 &q) const
+{
+    return ((x() * q.x()) + (y() * q.y()) + (z() * q.z()));
+}
+
+Vector3 Vector3::cross(const Vector3 &q) const
+{
+    Vector3 r;
```

Comment:
Better use the x, y, z constructor to avoid constructing and then rewriting the object.